packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -236,6 +237,24 @@ export class BuildService {
     return this.prisma.build.findUnique(args);
   }
 
+  async findByRunId(runId: string): Promise<Build | null> {
+    return this.prisma.build.findFirst({
+      where: {
+        runId
+      }
+    });
+  }
+
+  async logGenerateStatusByRunId(runId: string, status: string): Promise<void> {
+    const build = await this.findByRunId(runId);
+    const steps = await this.actionService.getSteps(build.actionId)
+    const generateStep = steps.find(step => step.name === GENERATE_STEP_NAME);  
```

Comment:
I see we are comparing `step.name` to a constant. I also see we have 3 step names defined in this file (GENERATE_APPLICATION, BUILD_DOCKER, PUSH_TO_GITHUB).

Can we please create an Enum to contain these types? For example, `CodeGenerationSteps`?

Reply:
After checking required changes in order to transfer usages to enum, I've decided to open an issue and solve it separately.